app/src/main/java/atlant/moviesapp/activity/MovieDetailsActivity.java

```diff
@@ -79,32 +82,88 @@
     @BindView(R.id.play_button)
     ImageView playButton;
 
+    @BindView(R.id.heart_detail)
+    ImageView favourite;
+
+    @BindView(R.id.bookmark_detail)
+    ImageView watchlist;
+
     @OnClick(R.id.play_button)
     public void showTrailer() {
         Bundle bundle = new Bundle();
-        bundle.putInt("id", movie.getId());
+        bundle.putInt(getString(R.string.id), movie.getId());
         YouTubeFragment f = new YouTubeFragment();
         f.setArguments(bundle);
         FragmentTransaction ft = getSupportFragmentManager().beginTransaction();
         ft.add(R.id.main, f, f.getTag());
-        // ft.replace(R.id.container, f, f.getTag());
         ft.commit();
     }
 
     @BindView(R.id.rate_txtBtn)
     TextView rateTxt;
     private static final int TAG = 0;
+    private BodyFavourite bodyFavourite;
+    private BodyWatchlist bodyWatchlist;
 
     @OnClick(R.id.rate_txtBtn)
     void rate() {
         Intent i = new Intent(this, RatingActivity.class);
-        i.putExtra("title", "Rate this movie");
-        i.putExtra("id", movie.getId());
-        i.putExtra("tag", TAG);
+        i.putExtra(getString(R.string.title), getString(R.string.rateThisMovie));
+        i.putExtra(getString(R.string.id), movie.getId());
+        i.putExtra(getString(R.string.tag), TAG);
         startActivity(i);
 
     }
 
+    @OnClick(R.id.heart_detail)
+    void addToFavorites() {
+        if (ApplicationState.getUser().getFavouriteMovies().contains(movie.getId())) {
+            ApplicationState.getUser().removeFavouriteMovie(movie.getId());
+            bodyFavourite = new BodyFavourite(getString(R.string.movie), movie.getId(), false);
+            presenter.postFavorite(movie.getId(), ApplicationState.getUser().getSessionId(), bodyFavourite);
+            Toast.makeText(this, R.string.removedFavorite, Toast.LENGTH_SHORT).show();
+            Glide.with(this).load(R.drawable.like)
+                    .crossFade().centerCrop()
+                    .diskCacheStrategy(DiskCacheStrategy.ALL)
+                    .into(favourite);
+
+        } else {
+            ApplicationState.getUser().addFavouriteMovie(movie.getId());
+            bodyFavourite = new BodyFavourite(getString(R.string.movie), movie.getId(), true);
+            presenter.postFavorite(movie.getId(), ApplicationState.getUser().getSessionId(), bodyFavourite);
+            Toast.makeText(this, R.string.addedFavorite, Toast.LENGTH_SHORT).show();
+            Glide.with(this).load(R.drawable.like_active_icon)
+                    .crossFade().centerCrop()
+                    .diskCacheStrategy(DiskCacheStrategy.ALL)
+                    .into(favourite);
+
+        }
+
+    }
+
+
+    @OnClick(R.id.bookmark_detail)
```

Comment:
Please rename method as it either removes or adds to the watchlist. example : updateWatchlist()